libcdoc/Tar.cpp

```diff
@@ -222,8 +222,15 @@ libcdoc::TarConsumer::open(const std::string& name, int64_t size)
 	_current_size = size;
 	Header h {};
 	std::string filename(name);
-	std::string filenameTruncated(filename.begin(), filename.begin() + h.name.size());
-	std::copy(filenameTruncated.cbegin(), filenameTruncated.cend(), h.name.begin());
+    size_t lastSlashPos = name.find_last_of("\\/");
+    if (lastSlashPos != std::string::npos)
+    {
+        std::copy(name.cbegin() + lastSlashPos + 1, name.cend(), h.name.begin());
+    }
+    else
+    {
+        std::copy(name.cbegin(), name.cend(), h.name.begin());
+    }
```

Comment:
I think we should allow multilevel path names in container just in case someone want to encrypt a directory tree